util/settings/settings.go

```diff
@@ -880,14 +880,45 @@ func (a *ArgoCDSettings) checkResourcePresence(apiGroup, kind, cluster string, f
 	return false
 }
 
-func (a *ArgoCDSettings) IsIncludedResource(apiGroup, kind, cluster string) bool {
+func (a *ArgoCDSettings) isIncludedResource(apiGroup, kind, cluster string) bool {
 	return a.checkResourcePresence(apiGroup, kind, cluster, a.ResourceInclusions)
 }
 
-func (a *ArgoCDSettings) IsExcludedResource(apiGroup, kind, cluster string) bool {
+func (a *ArgoCDSettings) isExcludedResource(apiGroup, kind, cluster string) bool {
 	return a.checkResourcePresence(apiGroup, kind, cluster, a.getExcludedResources())
 }
 
-func (a *ArgoCDSettings) IsWhitelistAvailable() bool {
-	return len(a.ResourceInclusions) > 0
+// Behavior of this function is as follows:
+// +-------------+-------------+-------------+
+// |  Inclusions |  Exclusions |    Result   |
+// +-------------+-------------+-------------+
+// |    Empty    |    Empty    |   Allowed   |
+// +-------------+-------------+-------------+
+// |   Present   |    Empty    |   Allowed   |
+// +-------------+-------------+-------------+
+// | Not Present |    Empty    | Not Allowed |
+// +-------------+-------------+-------------+
+// |    Empty    |   Present   | Not Allowed |
+// +-------------+-------------+-------------+
+// |    Empty    | Not Present |   Allowed   |
+// +-------------+-------------+-------------+
+// |   Present   | Not Present |   Allowed   |
+// +-------------+-------------+-------------+
+// | Not Present |   Present   | Not Allowed |
+// +-------------+-------------+-------------+
+// | Not Present | Not Present | Not Allowed |
+// +-------------+-------------+-------------+
+// |   Present   |   Present   | Not Allowed |
+// +-------------+-------------+-------------+
+//
+func (a *ArgoCDSettings) IsExcludedResource(apiGroup, kind, cluster string) bool {
+	if len(a.ResourceInclusions) > 0 {
```

Comment:
I feel like this could probably be simpler. 

```golang
return a.isExcludedResource(apiGroup, kind, cluster) || !a.isIncludedResource(apiGroup, kind, cluster)
```

Reply:
I actually thought about this, but I went against this because of a scenario similar to Alex's above:

```
resource.inclusions: |
    - apiGroups:
      - apps
      kinds:
      - DaemonSet
resource.exclusions: |
    - apiGroups:
      - apps
```

When considering a resource with `apiGroups: [apps], kinds: [DaemonSet]`, in the simplified version the operand `a.isExcludedResource(apiGroup, kind, cluster)` would return true for it (it is still under `apiGroups: [apps]`) and the statement would short-circuit and return true. However, if it were allowed to be considered by `a.isIncludedResource(apiGroup, kind, cluster)` it would be allowed and the statement would return false (as expected).